File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/connection/IonHelper.java
package pl.tobzzo.miechowskiepowietrze.connection;

import com.google.gson.JsonObject;
import javax.inject.Inject;
import javax.inject.Singleton;
import kotlin.Unit;
import kotlin.jvm.functions.Function2;
import org.jetbrains.annotations.NotNull;


public class IonHelper {

  public IonHelper(){

  }

  public void getSmth(@NotNull String url, @NotNull String apiKey,
      @NotNull Function2<? super Exception, ? super JsonObject, Unit> function) {


  }
}

File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/connection/IonHelper.kt
package pl.tobzzo.miechowskiepowietrze.connection

import android.content.Context
import com.google.gson.JsonObject
import com.koushikdutta.ion.Ion


class IonHelper(private val context: Context) {

  fun getSmth(url: String, apiKey: String,
    callback: Function2<Exception?, JsonObject, Unit>) {
    Ion.with(context)
      .load(url)
      .addHeader("apikey", apiKey)
      .asJsonObject()
      .setCallback { e, result ->
        callback(e, result)
      }
  }
}
